content/docs/features/dockerfiles.md

```diff
@@ -0,0 +1,249 @@
+  prior to `build` when using extensions.
+
+### Phased approach
+
+Some limitations of the initial implementation of the Dockerfiles feature have already been mentioned, and we'll expand
+on them here. As this is a large and complicated feature, the implementation has been split into phases in order to
+deliver incremental value and gather feedback.
+
+* Phase 1: one or more `run.Dockerfile` can be used to switch (only) the run image; no image modifications are allowed (
+  this eliminates the need for an `extend` lifecycle phase)
+* Phase 2: one or more `build.Dockerfile` can be used to extend the builder image
+  * A new `extend` lifecycle phase is introduced to apply `build.Dockerfile`s from `generate` to the builder image
+* Phase 3: one or more `run.Dockerfile` can be used to extend the run image
+  * The `extend` lifecycle phase can be run in parallel for the builder and run images
+
+The final ordering of lifecycle phases will look something like the following:
+
+* `analyze`
+* `detect` - after standard detection, `detect` will also run extensions' `./bin/generate`; output Dockerfiles are
+  written to a volume
+* `extend` - applies one or more `build.Dockerfile` to the builder image
+* `extend` - applies one or more `run.Dockerfile` to the run image (could run in parallel with builder image extension)
+* `restore`
+* `build`
+* `export`
+
+The [pack cli](https://github.com/buildpacks/pack)
+and [Tekton task](https://github.com/tektoncd/catalog/tree/main/task/buildpacks-phases/0.2) have or will eventually have
+support for builds using image extensions. However, [kpack](https://github.com/pivotal/kpack) - while able to use image
+extensions in builds - will need additional updating to propagate changes such as a switched run image back to the
+appropriate resource to avoid breaking rebase.
+
+Note that the method of applying Dockerfiles is left up to the platform, but could utilize the Docker daemon if
+available, or [`kaniko`](https://github.com/GoogleContainerTools/kaniko) for containerized workflows. The CNB lifecycle
+will have an `extender` binary to cover one or possibly both cases.
+
+## In action: a CNB build with extensions
+
+Let's walk through a build that uses extensions, step by step. We will see an image extension that switches the run
+image from an image that does not have `curl` installed to an image that does have `curl` installed.
+
+### 0. Setup workspace directory
+
+* `workspace=<preferred workspace directory>`
+
+### 1. Clone the lifecycle repo and build it (TODO: remove when lifecycle v0.15.0-rc.1 released)
+
+* `cd $workspace`
+* `git clone git@github.com:buildpacks/lifecycle.git`
+* `cd lifecycle`
+* `make clean build-linux-amd64 package-linux-amd64`
+* `LIFECYCLE_TARBALL=$(ls out/lifecycle-*.tgz)` - used for `pack builder create`
+* `LIFECYCLE_IMAGE=$(make build-image-linux-amd64 | grep 'tag lifecycle:' | cut -d ' ' -f 12)` - used for `pack build`
+
```

Comment:
We should not land these docs until `lifecycle`, `pack` and `samples` changes have landed.  Until then, we might want to let someone know that they can't build amd64 lifecycle on an ARM64 laptop.

```suggestion
  - An amd64 lifecycle image can only be built on a 64-bit Intel or AMD host
```

Reply:
https://github.com/buildpacks/docs/pull/526 leaves it up to the user to choose the right artifact for their arch